### ISSUE TITLE
Type intersection on type with nested type breaks extra properties error

### ISSUE THREAD

**Issue by Emily** [Microsoft/TypeScript#29678]
**TypeScript Version:** 3.4.0-dev.20190131 (I spot checked versions since 3.0.1, all had this issue

**Search Terms:** intersection extra properties

**Code**

```ts
type Problematic = { one?: string } & { nested?: { two?: string } }
function problem(input: Problematic) {}

// nested containing bad property unexpectedly doesn't cause error
problem({
  nested: {
    two: 'hello',
    bad: 123
  }
})
```

**Expected behavior:**
`bad: 123` should cause an extra properties error.

**Actual behavior:**
`bad: 123` does not cause the extra properties error.

**Playground Link:** [expanded examples](https://www.typescriptlang.org/play/#src=type%20Merge%3CT%20extends%20object%3E%20%3D%20%7B%20%5BK%20in%20keyof%20T%5D%3A%20T%5BK%5D%20%7D%3B%0D%0Atype%20Problematic%20%3D%20%7B%20one%3F%3A%20string%20%7D%20%26%20%7B%20nested%3F%3A%20%7B%20two%3F%3A%20string%20%7D%20%7D%0D%0Afunction%20problem(input%3A%20Problematic)%20%7B%7D%0D%0Afunction%20fine(input%3A%20Merge%3CProblematic%3E)%20%7B%7D%0D%0A%0D%0A%2F%2F%20nested%20containing%20bad%20property%20unexpectedly%20doesn't%20cause%20error%0D%0Aproblem(%7B%0D%0A%20%20nested%3A%20%7B%0D%0A%20%20%20%20two%3A%20'hello'%2C%0D%0A%20%20%20%20bad%3A%20123%0D%0A%20%20%7D%0D%0A%7D)%0D%0A%0D%0A%2F%2F%20nested%20containing%20bad%20property%20unexpectedly%20doesn't%20cause%20error%0D%0Aproblem(%7B%0D%0A%20%20one%3A%20'hello'%2C%0D%0A%20%20nested%3A%20%7B%0D%0A%20%20%20%20two%3A%20'hello'%2C%0D%0A%20%20%20%20bad%3A%20123%0D%0A%20%20%7D%0D%0A%7D)%0D%0A%0D%0A%2F%2F%20nested%20containing%20bad%20property%20with%20bad%20property%20at%20top-level%20_does_%20cause%20error%20on%20nested%0D%0A%2F%2F%20but%20weirdly%20not%20on%20the%20top-level%0D%0Aproblem(%7B%0D%0A%20%20bad%3A%20'test'%2C%0D%0A%20%20nested%3A%20%7B%0D%0A%20%20%20%20two%3A%20'hello'%2C%0D%0A%20%20%20%20bad%3A%20123%0D%0A%20%20%7D%0D%0A%7D)%0D%0A%0D%0A%2F%2F%20bad%20property%20at%20top-level%20without%20nested%20object%20does%20cause%20error%2C%20as%20expected%0D%0Aproblem(%7B%0D%0A%20%20bad%3A%20123%0D%0A%7D)%0D%0A%0D%0A%2F%2F%20examples%20where%20it%20seems%20to%20work%20more%20or%20less%20as%20expected%0D%0Afine(%7B%0D%0A%20%20nested%3A%20%7B%0D%0A%20%20%20%20two%3A%20'hello'%2C%0D%0A%20%20%20%20bad%3A%20123%0D%0A%20%20%7D%0D%0A%7D)%0D%0A%0D%0Afine(%7B%0D%0A%20%20one%3A%20'hello'%2C%0D%0A%20%20nested%3A%20%7B%0D%0A%20%20%20%20two%3A%20'hello'%2C%0D%0A%20%20%20%20bad%3A%20123%0D%0A%20%20%7D%0D%0A%7D)%0D%0A%0D%0Afine(%7B%0D%0A%20%20bad%3A%20123%2C%0D%0A%20%20nested%3A%20%7B%0D%0A%20%20%20%20two%3A%20'hello'%2C%0D%0A%20%20%20%20bad%3A%20123%0D%0A%20%20%7D%0D%0A%7D)%0D%0A%0D%0Afine(%7B%0D%0A%20%20bad%3A%20123%0D%0A%7D))

**Related Issues:** https://github.com/Microsoft/TypeScript/issues/28642

For anyone tripped up by this, there's a convenient work-around:
```ts
type Merge<T extends object> = { [K in keyof T]: T[K] };
type Problematic = { one?: string } & { nested?: { two?: string } }
function problemGone(input: Merge<Problematic>) {}

// works as expected
problemGone({
  nested: {
    two: 'hello',
    bad: 123
  }
})
```
